src/main/groovy/net/vivin/gradle/versioning/VersionUtils.groovy

```diff
@@ -140,7 +144,37 @@ class VersionUtils {
             refresh()
         }
 
-        return versions?.find()
+        if(!versionByTag) {
+            return null
+        } else {
+            def revWalk = new RevWalk(repository)
+            def candidateTags = []
+            def toInvestigateForTags = []
+
+            try {
+                toInvestigateForTags.add(revWalk.parseCommit(repository.resolve(Constants.HEAD)))
+
+                for(RevCommit investigatee = toInvestigateForTags.pop();
```

Comment:
oh - it took me a while to parse this haha! I don't think I've seen a `for` written quite like this before!